mithril-common/benches/merkle_map.rs

```diff
@@ -0,0 +1,241 @@
+
+fn generate_block_ranges_nodes_iterator(
+    total_transactions: u64,
+    total_transactions_per_block: u64,
+    block_range_length: u64,
+    max_uncompressed_block_ranges: u64,
+) -> impl Iterator<Item = (BlockRange, MKMapNode<BlockRange>)> {
+    let total_block_ranges =
+        total_transactions / (total_transactions_per_block * block_range_length);
+    assert!(
+        total_block_ranges > 0,
+        "total_block_ranges should be strictly greater than 0"
+    );
+    (0..total_block_ranges).map(move |block_range_index| {
+        let block_range = BlockRange::new(
+            block_range_index * total_transactions_per_block * block_range_length,
+            (block_range_index + 1) * total_transactions_per_block * block_range_length,
+        );
+        let mk_map_node = if block_range_index < max_uncompressed_block_ranges {
+            let leaves = <Range<u64> as Clone>::clone(&block_range)
+                .map(|leaf_index| leaf_index.to_string())
+                .collect::<Vec<_>>();
+            let merkle_tree_block_range = MKTree::new(&leaves).unwrap();
+            merkle_tree_block_range.into()
+        } else {
+            MKMapNode::TreeNode(block_range.to_string().into())
+        };
+
+        (block_range, mk_map_node)
+    })
+}
+
+fn generate_merkle_map_compressed(
+    block_ranges_nodes_iterator: impl Iterator<Item = (BlockRange, MKMapNode<BlockRange>)>,
+) -> MKMap<BlockRange, MKMapNode<BlockRange>> {
+    let mut mk_map = MKMap::new(&[]).unwrap();
+    for (block_range, mk_map_node) in block_ranges_nodes_iterator {
+        mk_map
+            .insert(block_range, mk_map_node.compute_root().unwrap().into())
+            .unwrap();
+    }
+    mk_map
+}
+
+fn generate_merkle_map_root(
+    block_ranges_nodes_iterator: impl Iterator<Item = (BlockRange, MKMapNode<BlockRange>)>,
+    mk_map_compressed: &MKMap<BlockRange, MKMapNode<BlockRange>>,
+) -> MKMapProof<BlockRange> {
+    let (mk_map_key_to_prove, mk_map_node_to_prove) =
+        &block_ranges_nodes_iterator.take(1).collect::<Vec<_>>()[0];
+    let mktree_to_prove = if let MKMapNode::Tree(mktree_to_prove) = mk_map_node_to_prove {
+        mktree_to_prove
+    } else {
+        panic!("Expected MKMapNode::Tree");
+    };
+    let leaves_to_prove = mktree_to_prove
+        .leaves()
+        .into_iter()
+        .take(1)
+        .collect::<Vec<_>>();
+    let mut mk_map_compressed = mk_map_compressed.clone();
+    mk_map_compressed
+        .insert(
+            mk_map_key_to_prove.to_owned(),
+            mk_map_node_to_prove.to_owned(),
+        )
+        .unwrap();
+    mk_map_compressed.compute_proof(&leaves_to_prove).unwrap()
+}
+
+fn generate_merkle_map_proof(
+    block_ranges_nodes_iterator: impl Iterator<Item = (BlockRange, MKMapNode<BlockRange>)>,
+    mk_map_compressed: &MKMap<BlockRange, MKMapNode<BlockRange>>,
+    total_proofs: u64,
+) -> MKMapProof<BlockRange> {
+    let mut leaves_to_prove_all: Vec<MKTreeNode> = vec![];
```

Comment:
This function is very similar to `generate_merkle_map_root`. Moving line 87 upper and line 118 lower highlights similarity. I have the impression that we do the same thing on list and on one element (which is a list of one element)